### DISCUSSION TITLE
Server Error

### DISCUSSION THREAD

**Discussion by hemeanandh** [vercel/next.js#36070] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64
    Binaries:
      Node: 14.17.5
      npm: 6.14.14
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.0
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome, MacOs 12.3.1

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

I have an app written in nextjs(running in 3000). I have a proxy configured for it(running in 3001). Any unknown route goes to proxy app. 
I have a local set up for the app. I am able to see the main application running well. All the pages are served. But when I redirect to a proxy route i can only see below error
Server Error
Error: connect ECONNREFUSED ::1:3001

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
<unknown> (Error: connect ECONNREFUSED
: (1:3001)
TCPConnectWrap.afterConnect [as oncomplete]
node:net (1195:16)

This problem only appears to me. All my colleagues have same setup(same packages, same version, same nodeversion). But still only I get to see the issue. Also the deployed app looks good. I face this issue in local only
Kindly help

### Expected Behavior

The application should run in local environment

### To Reproduce

<img width="1030" alt="image" src="https://user-images.githubusercontent.com/16472908/162789603-c6136078-db26-4417-8e36-0b2e3c5c5855.png">

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think this might be a networking issue rather than a Next.js problem, seeing that you have `::1:3001`, which is, AFAIK, an IPv6.

- Are you perhaps using Firefox? I do recall having problems with an auth service that required the IP of the client, but couldn't account for "IPv4 mapped IPv6" IPs,  which was only sent by FF. Chrome, Safari, etc., did send the IPv4 IP.
  - I just saw you use Chrome, but it'd still check that your local network
- Maybe try replacing `localhost` for `127.0.0.1` in your proxy.

**Reply by hemeanandh** [reply]
Thanks..Replacing localhost:3001 with 127.0.0.1:3001 worked

**Reply by msk-psp** [reply]
I experienced same thing. but It was works before installing Docker. 

I don't know why, maybe Docker changed some configurations about network.

However AFAIK localhost can be resolved into IPv4, IPv6 while 127.0.0.1 is not, how to check the network settings ? I'm using macOS now

**Reply by RTiwary** [reply]
this worked for me too, but does anyone know why this is? kinda confused. I get that 127.0.0.1 is an IPV4 address and ::1 is IPV6, but what does that have to do with this?
